node/services/service.go

```diff
@@ -3,32 +3,73 @@ package services
 import (
 	"context"
 
+	"github.com/ipfs/go-datastore"
 	ipld "github.com/ipfs/go-ipld-format"
 	"github.com/ipfs/go-merkledag"
+	"github.com/libp2p/go-libp2p-core/host"
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	"go.uber.org/fx"
 
+	"github.com/celestiaorg/celestia-node/core"
 	"github.com/celestiaorg/celestia-node/das"
 	"github.com/celestiaorg/celestia-node/node/fxutil"
 	"github.com/celestiaorg/celestia-node/service/block"
 	"github.com/celestiaorg/celestia-node/service/header"
 	"github.com/celestiaorg/celestia-node/service/share"
 )
 
-// Header constructs a new header.Service.
-func Header(lc fx.Lifecycle, ps *pubsub.PubSub) (*header.Service, header.Broadcaster) {
-	service := header.NewHeaderService(nil, nil, ps)
+// HeaderSyncer creates a new header.Syncer.
+func HeaderSyncer(cfg Config) func(ex header.Exchange, store header.Store) (*header.Syncer, error) {
+	return func(ex header.Exchange, store header.Store) (*header.Syncer, error) {
+		genesis, err := cfg.genesis()
+		if err != nil {
+			return nil, err
+		}
+
+		return header.NewSyncer(ex, store, genesis), nil
+	}
+}
+
+// HeaderService creates a new header.Service.
+func HeaderService(lc fx.Lifecycle, syncer *header.Syncer, sub *pubsub.PubSub) (*header.Service, header.Broadcaster) {
```

Comment:
Should we just have / provide a separate function that takes in `*header.Service` and returns it as a `header.Broadcaster` ? It'll make it clearer.

Reply:
we can do this after devnet though.

Reply:
Yep